src/dsymbol.d

```diff
@@ -1366,7 +1366,12 @@ public:
                 if (!(flags & IgnoreErrors) && s.prot().kind == PROTprivate &&
                     !s.isOverloadable() && !s.parent.isTemplateMixin() && !s.parent.isNspace())
                 {
-                    if (!s.isImport())
+                    AliasDeclaration ad = void;
+                    // accessing private selective and renamed imports is
+                    // deprecated by restricting the symbol visibility
+                    if (s.isImport() || (ad = s.isAliasDeclaration()) !is null && ad._import !is null)
```

Comment:
The eager access checks during lookup are disabled for renamed and selective imports. This is instead handled by the new DIP22 symbolIsVisible check above.
This means that people will only get a deprecation warning for now (b/c of the additional IgnoreSymbolVisibility lookup), and that `-transition=import`/`-transition=checkimports` work as for any other private symbol w/ the DIP22 changes.